src/satosa/base.py

```diff
@@ -137,12 +136,18 @@ def _auth_resp_callback_func(self, context, internal_response):
         internal_response.requester = context.state[STATE_KEY]["requester"]
 
         # If configured construct the user id from attribute values.
-        if "user_id_from_attrs" in self.config["INTERNAL_ATTRIBUTES"]:
-            subject_id = [
-                "".join(internal_response.attributes[attr]) for attr in
-                self.config["INTERNAL_ATTRIBUTES"]["user_id_from_attrs"]
-            ]
-            internal_response.subject_id = "".join(subject_id)
+        try:
+            if "user_id_from_attrs" in self.config["INTERNAL_ATTRIBUTES"]:
+                subject_id = [
+                    "".join(internal_response.attributes[attr]) for attr in
+                    self.config["INTERNAL_ATTRIBUTES"]["user_id_from_attrs"]
+                ]
+                internal_response.subject_id = "".join(subject_id)
+        except KeyError as err:
```

Comment:
otherwise if a misconfiguration would happen SATOSA returns a KeyError exception from which it is difficult to know the cause from the logs.

This littl change makes satosa just more eloquent